util/common/traces.go

```diff
@@ -112,3 +112,13 @@ func SegmentValidationTest(t *testing.T, traceTest TraceTestConfig, segments []t
 	return nil
 
 }
+func GenerateTraces(traceTest TraceTestConfig) error {
+	CopyFile(traceTest.AgentConfigPath, ConfigOutputPath)
+	go func() {
+		traceTest.Generator.StartSendingTraces(context.Background())
+	}()
+	time.Sleep(traceTest.AgentRuntime)
+	traceTest.Generator.StopSendingTraces()
+	time.Sleep(AGENT_SHUTDOWN_DELAY)
+	return nil
```

Comment:
Why does `GenerateTraces` return an error when it's only ever nil? Same question for the other functions in this file.

Reply:
For the other function they fail if there is an error so if you reach the end there is not error.  I understand what you mean tho. Those function could be changed to void.